sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -232,11 +242,16 @@ case class AlterViewAsCommand(
 
   override def childrenToAnalyze: Seq[LogicalPlan] = query :: Nil
 
+  override def applyAnalysisContext(analysisContext: AnalysisContext): AnalysisOnlyCommand = {
+    // Collect the referred temporary functions from AnalysisContext
+    copy(referredTempFunctions = analysisContext.referredTempFunctionNames.toSeq)
+  }
+
   def markAsAnalyzed(): LogicalPlan = copy(isAnalyzed = true)
```

Comment:
i'm thinking about combining them to
```
def markAsAnalyzed(analysisContext: AnalysisContext): LogicalPlan = {
  copy(
    isAnalyzed = true,
    referredTempFunctions = analysisContext.referredTempFunctionNames.toSeq)
}
```

cannot decide which is better

Reply:
I think combining is better.